ADFSToolkit/Public/get-ADFSTkFederationDefaults.ps1

```diff
@@ -0,0 +1,168 @@
+﻿function get-ADFSTkFederationDefaults
+{
+ [CmdletBinding()]
+    param (
+        $URL,
+        [switch]$InstallDefaults,
+        $ExtractionFilter ="ADFSToolkit/config/federation",
+        [switch]$ClearCache,
+        [switch]$Silent
+    )
+
+process 
+{
+    #Prepare to be able to use zip files 
+    Add-Type -assembly "system.io.compression.filesystem"
+
+    #Get All paths  and assert they exist 
+       
+    $Global:ADFSTkPaths = Get-ADFSTKPaths
+        
+
+    # setup our files
+    $nowStamp=[DateTime]::Now.ToString("yyyyMMdd-HHmmss")
+    $fedCacheFile="federationdefaults.zip"
+    $fedCacheFileBackup="federationdefaults" +$nowStamp +".zip"
+    $federationDirNameBackup="federation"+$nowStamp
+ 
+    $federationConfigCacheFile=         Join-Path $Global:ADFSTkPaths.cacheDir $fedCacheFile
+    $federationConfigCacheFileBackup=   Join-Path $Global:ADFSTkPaths.cacheDir $fedCacheFileBackup
+    $federationConfigDirFullBackupPath= Join-Path $Global:ADFSTkPaths.mainConfigDir $federationDirNameBackup
```

Comment:
We should have a backupdir  in the federations folder instead.
Please add a line inte Get-ADFStkPaths named federationBackupDir: 
`$paths.federationBackupDir = Join-Path $paths.federationDir 'backup'`
and use that in the code.